server/bin/pbench-unpack-tarballs.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+import os
+import sys
+import glob
+import tempfile
+
+from logging import Logger
+from pathlib import Path
+
+from pbench.common.exceptions import BadConfig
+from pbench.common.logger import get_pbench_logger
+from pbench.server import PbenchServerConfig
+from pbench.server.filetree import Controller
+from pbench.server.report import Report
+from pbench.server.database import init_db
+
+
+_NAME_ = "pbench-unpack-tarballs"
+
+# The link source for the operation of this script.
+_linksrc = "TO-UNPACK"
+
+
+def sanity_check(config: PbenchServerConfig, logger: Logger):
+    try:
+        incoming_p = config.INCOMING.resolve(strict=True)
+    except FileNotFoundError:
+        logger.exception(
+            f"The configured INCOMING directory, {config.INCOMING}, does not exist",
+        )
+        incoming_p = None
+    else:
+        if not incoming_p.is_dir():
+            logger.error(
+                f"The configured INCOMING directory, {config.INCOMING}, is not a valid directory",
+            )
+            incoming_p = None
+
+    try:
+        results_p = config.RESULTS.resolve(strict=True)
+    except FileNotFoundError:
+        logger.exception(
+            f"The configured RESULTS directory, {config.RESULTS}, does not exist",
+        )
+        results_p = None
+    else:
+        if not results_p.is_dir():
+            logger.error(
+                f"The configured RESULTS directory, {config.RESULTS}, is not a valid directory",
+            )
+            results_p = None
+
+    return incoming_p, results_p
+
+
+def unpack_tarball(
+    config: PbenchServerConfig, logger: Logger, incoming: Path, results: Path
+):
+
+    tarlist = glob.iglob(os.path.join(config.ARCHIVE, "*", _linksrc, "*.tar.xz"))
+    ntotal = nsuccess = 0
+
+    for tarball in sorted(tarlist):
+        ntotal += 1
+
+        try:
+            tb = Path(tarball).resolve(strict=True)
+        except FileNotFoundError:
+            logger.error(
+                "Tarball link, '{}', does not resolve to a real location", tarball
+            )
+
+        controller_path = tb.parent
+        controller = Controller.create(controller_path.name, config, logger)
+        tar = controller.create_tarball(tb)
+        try:
+            tar.unpack(incoming, results)
+        except Exception:
```

Comment:
You're currently not raising an exception in `unpack` if something goes wrong, although I think you should!